lib/rubocop/cop/style/fetch_env_var.rb

```diff
@@ -69,9 +69,22 @@ def message_chained_with_dot?(node)
         # - Used as a flag (e.g., `if ENV['X']` or `!ENV['X']`) because
         #   it simply checks whether the variable is set.
         # - Receiving a message with dot syntax, e.g. `ENV['X'].nil?`.
-        # - `ENV['key']` is a receiver of `||=`, e.g. `ENV['X'] ||= y`.
         def allowable_use?(node)
-          used_as_flag?(node) || message_chained_with_dot?(node) || node.parent&.or_asgn_type?
+          used_as_flag?(node) || message_chained_with_dot?(node) || assigned?(node)
+        end
+
+        # The following are allowed cases:
+        #
+        # - `ENV['key']` is a receiver of `||=`, e.g. `ENV['X'] ||= y`.
+        # - `ENV['key']` is a receiver of `&&=`, e.g. `ENV['X'] &&= y`.
+        def assigned?(node)
+          return false unless node.parent&.assignment?
+
+          node.parent.each_child_node.inject(true) do |last_not_env_node, child_node|
+            break true unless last_not_env_node
```

Comment:
The specs seems to pass without this `break`. What about the following logic?

```diff
diff --git a/lib/rubocop/cop/style/fetch_env_var.rb b/lib/rubocop/cop/style/fetch_env_var.rb
index 64f5b81d8..e9c5e375b 100644
--- a/lib/rubocop/cop/style/fetch_env_var.rb
+++ b/lib/rubocop/cop/style/fetch_env_var.rb
@@ -78,13 +78,11 @@ module RuboCop
         # - `ENV['key']` is a receiver of `||=`, e.g. `ENV['X'] ||= y`.
         # - `ENV['key']` is a receiver of `&&=`, e.g. `ENV['X'] &&= y`.
         def assigned?(node)
-          return false unless node.parent&.assignment?
+          return false unless (parent = node.parent)&.assignment?

-          node.parent.each_child_node.inject(true) do |last_not_env_node, child_node|
-            break true unless last_not_env_node
+          lhs, _method, _rhs = *parent

-            child_node != node
-          end
+          node == lhs
         end
       end
     end
```

Reply:
Thank you so much! It is very simple.